google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerImpl.java

```diff
@@ -263,18 +260,14 @@ public DatabaseClient getDatabaseClient(DatabaseId db) {
       }
       dbClients.clear();
     }
-    return new ListenableFutureToApiFuture<>(transform(
-        Futures.successfulAsList(closureFutures),
-        new Function<List<Void>, Void>() {
-          @Override
-          public Void apply(List<Void> inputs) {
-            for (ManagedChannel channel : getOptions().getRpcChannels()) {
-              channel.shutdown();
-            }
-            return null;
-          }
-        },
-        directExecutor()));
+    try {
+      Futures.successfulAsList(closureFutures).get();
+    } catch (InterruptedException | ExecutionException e) {
+      throw SpannerExceptionFactory.newSpannerException(e);
+    }
+    for (ManagedChannel channel : getOptions().getRpcChannels()) {
+      channel.shutdown();
```

Comment:
Should the whole process fail in one channel.shutdown() fails (throws exception)?